examples/camera/screen_shake.rs

```diff
@@ -0,0 +1,147 @@
+        material: materials.add(Color::srgb(0.2, 0.2, 0.3)),
+        ..default()
+    });
+
+    // Player
+    commands.spawn((
+        Player,
+        MaterialMesh2dBundle {
+            mesh: meshes.add(Circle::new(25.)).into(),
+            material: materials.add(Color::srgb(6.25, 9.4, 9.1)), // RGB values exceed 1 to achieve a bright color for the bloom effect
+            transform: Transform {
+                translation: vec3(0., 0., 2.),
+                ..default()
+            },
+            ..default()
+        },
+    ));
+
+    // Add screen shake component to track shake duration and intensity
+    commands.insert_resource(ScreenShake::new(10.0, 1.0));
+}
+
+fn setup_instructions(mut commands: Commands) {
+    commands.spawn(
+        TextBundle::from_section("Use space to trigger a screen shake", TextStyle::default())
+            .with_style(Style {
+                position_type: PositionType::Absolute,
+                bottom: Val::Px(12.0),
+                left: Val::Px(12.0),
+                ..default()
+            }),
+    );
+}
+
+fn setup_camera(mut commands: Commands) {
+    commands.spawn((
+        Camera2dBundle {
+            camera: Camera {
+                hdr: true, // HDR is required for the bloom effect
+                ..default()
+            },
+            ..default()
+        },
+        Bloom::NATURAL,
+    ));
+}
+
+#[derive(Resource)]
+struct ScreenShake {
+    intensity: f32,
+    duration: f32,
+    timer: Timer,
+}
+
+impl ScreenShake {
+    fn new(intensity: f32, duration: f32) -> Self {
+        ScreenShake {
+            intensity,
+            duration,
+            timer: Timer::from_seconds(0.0, TimerMode::Once), // Start timer at 0
+        }
+    }
+
+    fn start_shake(&mut self, intensity: f32, duration: f32) {
+        self.intensity = intensity;
+        self.duration = duration;
+        self.timer = Timer::from_seconds(duration, TimerMode::Once); // Reset the timer
+    }
+}
+
+fn trigger_shake_on_space(
+    keyboard_input: Res<ButtonInput<KeyCode>>,
+    mut screen_shake: ResMut<ScreenShake>,
+) {
+    if keyboard_input.just_pressed(KeyCode::Space) {
+        // Start the screen shake with intensity 10.0 for 1 second
+        screen_shake.start_shake(INTENSITY, DURATION);
+    }
+}
+
+fn screen_shake(
+    time: Res<Time>,
+    mut screen_shake: ResMut<ScreenShake>,
+    mut query: Query<&mut Transform, With<Camera>>,
+) {
+    if screen_shake.timer.finished() {
+        return; // No shake if the timer has finished
+    }
+
+    // Update the shake timer
+    screen_shake.timer.tick(time.delta());
+
+    // Calculate the progress of the shake (percentage of time passed)
+    let elapsed = screen_shake.timer.elapsed_secs();
+    let total_duration = screen_shake.timer.duration().as_secs_f32();
+    let shake_progress = elapsed / total_duration;
+
+    let mut rng = ChaCha8Rng::from_seed(42);
+    let shake_amount = screen_shake.intensity * (1.0 - shake_progress);
```

Comment:
you can use `fraction_remaining` her, it does `1.0 - (elapsed / remaining)` for you

```suggestion
    let mut rng = ChaCha8Rng::from_seed(42);
    let shake_amount = screen_shake.intensity * screen_shake.timer.fraction_remaining();
```